src/dmd/target.d

```diff
@@ -120,6 +129,7 @@ extern (C++) struct Target
     }
 
     OS os = defaultTargetOS();
+    uint freeBSDMajor = defaultFreeBSDVersion();
```

Comment:
Why can't you set the version condition by parsing the target triple on demand?

Reply:
n.b. this is the default.

Reply:
But it doesn't belong in Target

Reply:
(rhetorical) Then where does it belong? 

It is here for the same reason that `os` has a default initial value, because when both host and target are freebsd then the default version of freebsd to be targeted should be that of the host.

Reply:
Target is an abstract interface to describe a backend.  This freeBSDMajor is an internal implementation detail that should not pollute gdc and ldc.

Reply:
the question still stands. Where does this belong? 
Before it was a statically predetermined constant in an extern(D) function, now it is a dynamic variable.
LDC will use precisely none of this code as it is already a cross-compiler and has proper triple support from LLVM.

This is essentially OS major field (and thus I believer it should go next to the os field) that may be useful for MacOS at some point as there are a number os symbols that change version to version, but for the moment AFAIU FreeBSD is the only OS that breaks across versions in a way that can't be adapted for without different D versions for different OS versions.

Reply:
The target string is already floating around in `dmd.mars`, can't re-use the same string to get out the version when setting predefined versions?

`extern(D) static private` is also possible...

Reply:
Just for DMD, following their naming convention, the meat of `target.d` could be split up into `target_linux.d`, `target_freebsd.d`, `target_x86.d`, `target_arm.d`, etc...

Reply:
not without reparsing it